### DISCUSSION TITLE
How to access request data (evdata) from a worker thread?

### DISCUSSION THREAD

**Discussion by mcordova1967** [cesanta/mongoose#1682] [Ask questions, get answers] [answered]
I was able to run the multithreaded example, pretty straightforward, but I can't figure out how to access evdata from the thread in order to extract information from the request that the task would need to perform its work, something like this:

struct mg_http_message *hm = (struct mg_http_message *) ev_data;

The actual thread function only receives a socket:

static void thread_function(void *param) 

Thank you.

**Comment by scaprile** [comment, marked as answer]
[from the docs](https://mongoose.ws/documentation/#mg_mkpipe): 

> Also, `fn` can send any data to a worker thread, which can be `recv()`ed by a worker thread

just `mg_send(c->fn_data, ...);` in `fn` and `int n = recv(sock, buffer, sizeof(buffer), 0);  // THIS WILL BLOCK` in your worker thread

**Reply by scaprile** [reply]
OTOH, the thread worker receives a `void *`, perhaps it is simpler to just pass a structure containing the socket and the relevant data for the worker thread. We've just updated the example to do this.

**Reply by mcordova1967** [reply]
Thank you very much
